server/test/routes/login.js

```diff
@@ -0,0 +1,50 @@
+const test = require('tape');
+const supertest = require('supertest');
+
+const app = require('../../app');
+
+test('Testing login route with valid username and password', (t) => {
+  supertest(app)
+    .post('/api/v1/login')
+    .send({ email: 'ahmedisam9922@gmail.com', password: '123' })
+    .expect(200)
+    .expect('Content-Type', 'application/json; charset=utf-8')
+    .end((err, res) => {
+      if (err) t.error(err);
+      const resUser = JSON.parse(res.text);
```

Comment:
Instead of parsing the response text, you could test the body. 
```javascript 
const resUser = res.body
```
what you did is okay though :grin:

Reply:
You could also destructure it this way:
```javascript
const { first_name, last_name, address} = res.body
```